proto/data.go

```diff
@@ -546,3 +546,91 @@ type Int32Slice []int32
 func (s Int32Slice) Len() int           { return len(s) }
 func (s Int32Slice) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
 func (s Int32Slice) Less(i, j int) bool { return s[i] < s[j] }
+
+// ToInternal places the datapoints in a TimeSeriesData message into one or
+// more InternalTimeSeriesData messages. The structure and number of messages
+// returned depends on two variables: a key duration, and a sample duration.
+//
+// The sample duration is an interval length in nanoseconds; datapoints which
+// fall into the same interval will be aggregated together into a single Sample.
+//
+// The key duration is also a length in nanoseconds, which is used to determine
+// how many intervals are grouped into a single InternalTimeSeriesData
+// message. This value must be a multiple of the sample duration.
+//
+// Example: Assume the desired result is to aggregate individual datapoints into
+// the same sample if they occurred within the same second; additionally, all
+// samples which occur within the same hour should be stored at the same key
+// location within the same InternalTimeSeriesValue. The sample duration should
+// be 10^9 nanoseconds (value of time.Second), and the key duration should be
+// (3600*10^9) nanoseconds (value of time.Hour).
+//
+// Note that this method does not accumulate data into individual samples in the
+// case where multiple datapoints fall into the same sample period. Internal
+// data should be merged into the cockroach data store before being read; the
+// engine is responsible for accumulating samples.
+//
+// For more information on how time series data is stored, see
+// InternalTimeSeriesData and its related structures.
+func (ts TimeSeriesData) ToInternal(keyDuration int64, sampleDuration int64) (
+	[]*InternalTimeSeriesData, error) {
+	if keyDuration%sampleDuration != 0 {
+		return nil, util.Errorf(
```

Comment:
I don't think you need to pedantically write "error converting TimeSeriesData to InternalTimeSeriesData" with each error message.